processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -1086,18 +1091,72 @@ private Column deserializeColumn(ObjectMapper mapper, ByteBuffer byteBuffer) thr
     }
   }
 
+  public static String makeDimFileName(String dimension)
+  {
+    return String.format("dim_%s.drd", sanitizeFileName(dimension));
+  }
+
   public static File makeDimFile(File dir, String dimension)
   {
-    return new File(dir, String.format("dim_%s.drd", dimension));
+    return new File(dir, makeDimFileName(dimension));
+  }
+
+  public static String makeTimeFileName(ByteOrder order)
+  {
+    return String.format("time_%s.drd", order);
   }
 
   public static File makeTimeFile(File dir, ByteOrder order)
   {
-    return new File(dir, String.format("time_%s.drd", order));
+    return new File(dir, makeTimeFileName(order));
+  }
+
+  public static String makeMetricFileName(String metricName, ByteOrder order)
+  {
+    return String.format("met_%s_%s.drd", sanitizeFileName(metricName), order);
   }
 
   public static File makeMetricFile(File dir, String metricName, ByteOrder order)
   {
-    return new File(dir, String.format("met_%s_%s.drd", metricName, order));
+    return new File(dir, makeMetricFileName(metricName, order));
+  }
+
+  public static String sanitizeFileName(String fileName)
+  {
+    String encoding = System.getProperty("file.encoding", "UTF-8");
```

Comment:
this is not guaranteed to return the encoding used to encode the filename. I ran the following scala snippet on OS X 10.11, with `export LANG=C`. The resulting file name still uses a two-byte character, even though the byte length returns 1 and the encoding is set to US-ASCII. Basically, we can't infer anything from this encoding.

```
val c = "\u00e9"
val name = c + ".txt"

new java.io.FileOutputStream(name).write('A')
val charset = java.nio.charset.Charset.defaultCharset()
println(charset)
println(c.getBytes(charset).length)
```